pkg/shardddl/optimism/keeper_test.go

```diff
@@ -391,3 +393,63 @@ func (t *testKeeper) TestTargetTablesForTask(c *C) {
 			}),
 	})
 }
+
+func (t *testKeeper) TestRebuildLocksAndTables(c *C) {
+	var (
+		lk               = NewLockKeeper()
+		task             = "task"
+		source1          = "mysql-replica-1"
+		source2          = "mysql-replica-2"
+		upSchema         = "foo"
+		upTable          = "bar"
+		downSchema       = "db"
+		downTable        = "tbl"
+		DDLs1            = []string{"ALTER TABLE bar ADD COLUMN c1 INT"}
+		DDLs2            = []string{"ALTER TABLE bar DROP COLUMN c1"}
+		p                = parser.New()
+		se               = mock.NewContext()
+		tblID      int64 = 111
+		ti0              = createTableInfo(c, p, se, tblID, `CREATE TABLE bar (id INT PRIMARY KEY)`)
+		ti1              = createTableInfo(c, p, se, tblID, `CREATE TABLE bar (id INT PRIMARY KEY, c1 INT)`)
+		ti2              = createTableInfo(c, p, se, tblID, `CREATE TABLE bar (id INT PRIMARY KEY, c1 INT, c2 INT)`)
+		ti3              = createTableInfo(c, p, se, tblID, `CREATE TABLE bar (id INT PRIMARY KEY, c2 INT)`)
+
+		i11 = NewInfo(task, source1, upSchema, upTable, downSchema, downTable, DDLs1, ti0, []*model.TableInfo{ti1})
+		i21 = NewInfo(task, source2, upSchema, upTable, downSchema, downTable, DDLs2, ti2, []*model.TableInfo{ti3})
+
+		tts = []TargetTable{
```

Comment:
"target" in `TargetTable` makes me think of "source/target", where the "target" means downstream tables. We could rename it to "TableMap" or something in future